geowebcache/core/src/main/java/org/geowebcache/util/ResponseUtils.java

```diff
@@ -0,0 +1,319 @@
+/**
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * @author Sandro Salari, GeoSolutions S.A.S., Copyright 2017
+ * 
+ */
+
+package org.geowebcache.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.util.Arrays;
+import java.util.Date;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.httpclient.util.DateParseException;
+import org.apache.commons.httpclient.util.DateUtil;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geowebcache.GeoWebCacheDispatcher;
+import org.geowebcache.GeoWebCacheException;
+import org.geowebcache.conveyor.Conveyor;
+import org.geowebcache.conveyor.Conveyor.CacheResult;
+import org.geowebcache.conveyor.ConveyorTile;
+import org.geowebcache.filter.request.RequestFilterException;
+import org.geowebcache.filter.security.SecurityDispatcher;
+import org.geowebcache.grid.BoundingBox;
+import org.geowebcache.grid.GridSubset;
+import org.geowebcache.grid.OutsideCoverageException;
+import org.geowebcache.io.ByteArrayResource;
+import org.geowebcache.io.Resource;
+import org.geowebcache.layer.TileLayer;
+import org.geowebcache.layer.TileLayerDispatcher;
+import org.geowebcache.mime.ImageMime;
+import org.geowebcache.stats.RuntimeStats;
+import org.geowebcache.storage.DefaultStorageFinder;
+import org.springframework.http.MediaType;
+/**
+ * Utility methods that can be used to write a string as http response</br>
+ * The response can be a valid one or contains an error status code</br>
+ * The HTTP response can be rendered as HTML or XML
+ * @author sandr
+ *
+ */
+public final class ResponseUtils {
+
+    private static Log log = LogFactory.getLog(ResponseUtils.class);
+
+    private ResponseUtils() {
+    }
+
+    public static void writeTile(SecurityDispatcher secDispatcher, Conveyor conv, String layerName,
```

Comment:
Public method without javadoc. Several others in this class too.

Reply:
Fixed.